librz/util/skyline.c

```diff
@@ -10,7 +10,7 @@
 	(((addr) < (rz_itv_end(((RzSkylineItem *)(part))->itv)) || !rz_itv_end(((RzSkylineItem *)(part))->itv)) ? -1 : 1)
 
 RZ_API bool rz_skyline_add(RzSkyline *skyline, RzInterval itv, void *user) {
-	rz_return_val_if_fail(skyline, false);
+	rz_return_val_if_fail(skyline && (!rz_itv_size(itv) || !UT64_ADD_OVFCHK(rz_itv_begin(itv), rz_itv_size(itv) - 1)), false);
```

Comment:
I think it's probably to extract this long check into multiple `rz_return_val_if_fail()` for readability.